apollo-ios/Sources/Apollo/FieldPolicyEvaluator.swift

```diff
@@ -0,0 +1,190 @@
+import Foundation
+import ApolloAPI
+
+enum FieldPolicyResult {
+  case single(CacheKeyInfo)
+  case list([CacheKeyInfo])
+}
+
+struct FieldPolicyEvaluator {
+  let field: Selection.Field
+  let variables: GraphQLOperation.Variables?
+  
+  init(
+    field: Selection.Field,
+    variables: GraphQLOperation.Variables?
+  ) {
+    self.field = field
+    self.variables = variables
+  }
+  
+  func resolveFieldPolicy() -> FieldPolicyResult? {
+    guard let fieldPolicy = field.fieldPolicy,
+          let arguments = field.arguments else {
+      return nil
+    }
+    
+    struct ParsedKey { let name: String; let path: [String]? }
+
+    let parsed: [ParsedKey] = fieldPolicy.keys.map { key in
+      if let dot = key.firstIndex(of: ".") {
+        let name = String(key[..<dot])
+        let rest = key[key.index(after: dot)...]
+        return ParsedKey(name: name, path: rest.split(separator: ".").map(String.init))
+      } else {
+        return ParsedKey(name: key, path: nil)
+      }
+    }
+
+    var fixedParts = [String?](repeating: nil, count: parsed.count)
+    var listIndex: Int? = nil
+    var listValues: [String] = []
+
+    for (i, pk) in parsed.enumerated() {
```

Comment:
We try not to use 1 or 2 letter variable names. While reading this code, it took me an addition 15-20 seconds to figure out that `pk` stood for `parsedKey`. Making the code easier for others to quickly understand saves so much time and mental overhead in the long run. Just use `(index, parsedKey)`, or better yet `(index, keyArg)`.